source/cgal/intersections.cc

```diff
@@ -384,7 +384,7 @@ namespace CGALWrappers
             internal::mark_domains(cdt);
             std::array<Point<2>, 3> vertices;
 
-            for (const Face_handle &f : cdt.finite_face_handles())
+            for (const Face_handle f : cdt.finite_face_handles())
```

Comment:
Do we really need the explicit type change here, or would using `const auto &` work?

Reply:
> or would using `const auto &` work?

Potentially yes. But then I would be in favor of using `const auto &` in all loops of that kind in this file, i.e.
https://github.com/dealii/dealii/blob/2166009089bd1e3035d60d7d2b1fa0bbbbeb748d/source/cgal/intersections.cc#L151
https://github.com/dealii/dealii/blob/2166009089bd1e3035d60d7d2b1fa0bbbbeb748d/source/cgal/intersections.cc#L440

Reply:
I would say this is unrelated to the warnings fixed here.